backend-rust/src/graphql_api/baker_metrics.rs

```diff
@@ -0,0 +1,153 @@
+use crate::{
+    graphql_api::{get_pool, ApiError, ApiResult, MetricsPeriod},
+    scalar_types::{DateTime, TimeSpan},
+};
+use async_graphql::{Context, Object, SimpleObject};
+use chrono::Utc;
+use sqlx::postgres::types::PgInterval;
+use std::sync::Arc;
+
+#[derive(Default)]
+pub(crate) struct QueryBakerMetrics;
+
+#[Object]
+impl QueryBakerMetrics {
+    /// Fetches baker metrics for the specified period.
+    ///
+    /// This function queries the database for baker metrics such as the number
+    /// of bakers added, removed, and the last baker count in the specified
+    /// time period. It returns the results as a structured `BakerMetrics`
+    /// object.
+    async fn baker_metrics<'a>(
+        &self,
+        ctx: &Context<'a>,
+        period: MetricsPeriod,
+    ) -> ApiResult<BakerMetrics> {
+        let pool = get_pool(ctx)?;
+
+        let end_time = Utc::now();
+
+        let before_time = end_time - period.as_duration();
+
+        let before_period_row = sqlx::query!(
+            r#"
+            SELECT
+                total_bakers_added,
+                total_bakers_removed
+            FROM metrics_bakers
+            LEFT JOIN blocks ON metrics_bakers.block_height = blocks.height
+            WHERE blocks.slot_time < $1
+            ORDER BY metrics_bakers.block_height DESC
```

Comment:
I am not sure that postgres have enough information to be able to do this efficiently. Maybe I am wrong but it would need to know that `slot_time` and `block_height` are both strictly increasing, which is does not know. Maybe it would be better to sort by the `slot_time`:
```suggestion
            SELECT
                total_bakers_added,
                total_bakers_removed
            FROM metrics_bakers
            LEFT JOIN blocks ON metrics_bakers.block_height = blocks.height
            WHERE blocks.slot_time < $1
            ORDER BY blocks.slot_time DESC
```